code/modules/reagents/reagent_containers/food/drinks.dm

```diff
@@ -11,9 +23,27 @@
 	amount_per_transfer_from_this = 5
 	volume = 50
 	var/shaken = 0
+	var/drink_flags = NO_EMPTY_ICON
+	var/empty_icon_state = null
+
+/obj/item/reagent_containers/food/drinks/Initialize()
+	. = ..()
+	if(drink_flags & IS_GLASS)
+		unacidable = TRUE
 
 /obj/item/reagent_containers/food/drinks/on_reagent_change()
-	return
+	update_icon()
+
+/obj/item/reagent_containers/food/drinks/update_icon()
+	if(!reagents.total_volume)
+		if(drink_flags & UNIQUE_EMPTY_ICON)
+			icon_state = empty_icon_state
+			return
+		if(!(drink_flags & NO_EMPTY_ICON))
```

Comment:
```suggestion
		if(!(drink_flags & empty_icon_state))
```

Same here. Also, that ! also applies to the NO_EMPTY_ICON, so it does the reverse of what you intend it to do.

Reply:
No, that is not how flags work. This was tested. NO_EMPTY_ICON is required to avoid invisible sprites in this case, if something does not have an empty icon like you added with the bottles (ex: whiskeybottle_empty). This was why I had to bugfix some things like grenadine disappearing after the refactor.

Reply:
Ah I see